service-compiler/src/main/java/io/techery/janet/HttpHelpersFactoryGenerator.java

```diff
@@ -0,0 +1,103 @@
+package io.techery.janet;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeSpec;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.processing.Filer;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+
+import io.techery.janet.HttpActionService.ActionHelper;
+import io.techery.janet.compiler.utils.Generator;
+
+import static com.squareup.javapoet.MethodSpec.constructorBuilder;
+import static com.squareup.javapoet.MethodSpec.methodBuilder;
+import static com.squareup.javapoet.TypeSpec.classBuilder;
+
+public class HttpHelpersFactoryGenerator extends Generator<HttpActionClass> {
+
+    private final List<TypeElement> otherFactoriesElements;
+    private final Options options;
+
+    public HttpHelpersFactoryGenerator(Filer filer, List<TypeElement> otherFactories, Options options) {
+        super(filer);
+        this.otherFactoriesElements = otherFactories;
+        this.options = options;
+    }
+
+    @Override
+    public void generate(ArrayList<HttpActionClass> actionClasses) {
+        String className = HttpActionService.HELPERS_FACTORY_CLASS_SIMPLE_NAME;
+        if (options.factoryClassSuffix != null) className += options.factoryClassSuffix;
+
+        TypeSpec.Builder classBuilder = classBuilder(className)
+                .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+                .addJavadoc("Auto-generated class factory to create action helpers for actions")
+                .addSuperinterface(ParameterizedTypeName.get(HttpActionService.ActionHelperFactory.class));
+
+        MethodSpec.Builder constructorBuilder = constructorBuilder()
+                .addModifiers(Modifier.PUBLIC);
+
+        MethodSpec.Builder makeMethodBuilder = methodBuilder("make")
+                .addModifiers(Modifier.PUBLIC)
+                .addAnnotation(Override.class)
+                .returns(ActionHelper.class)
+                .addParameter(Class.class, "actionClass");
+
+        if (shouldIncludeOtherFactories()) {
+            // add field for other factories and initialize it in constructor
+            ParameterizedTypeName listType = ParameterizedTypeName.get(ArrayList.class, HttpActionService.ActionHelperFactory.class);
+            classBuilder.addField(
+                    FieldSpec.builder(listType, "otherFactories", Modifier.PRIVATE)
+                            .initializer("new $T()", listType)
+                            .build()
+            );
+            for (TypeElement factoryElement : otherFactoriesElements) {
+                if (factoryElement.getQualifiedName().equals(className)) continue;
```

Comment:
That was typo, sorry. I want to exclude canonical factory class by it's name. It could only happen if we generate general (no suffix) factory